[MRG+1] Fix segmentation fault in libsvm

lesteve opened the pull request:
Fix #6687.

The fix was taken from libsvm: 
https://github.com/cjlin1/libsvm/commit/745fdb02bf392b28a033921984901d24038472f7

I haven't really understood how to trigger this edge case in the libsvm code with some toy data, should I spend more time on this and try to write a test?

ogrisel commented:
We probably need a better sync mechanism with recent libsvm releases. It did not know that libsvm had a github repo. It used to be release tarball only without any VCS.

ogrisel commented:
@lesteve I think we should at least get the other fixes from cjlin1/libsvm@745fdb0 to make is simpler to keep the code in sync in the future.

Once done +1 for merge on my side.

lesteve committed:
FIX segmentation fault in libsvm

The fix was taken from libsvm:
https://github.com/cjlin1/libsvm/commit/745fdb02bf392b28a033921984901d24038472f7

lesteve commented:
> @lesteve I think we should at least get the other fixes from cjlin1/libsvm@745fdb0 to make is simpler to keep the code in sync in the future.

Done. Note the rest of the libsvm changes in cjlin1/libsvm@745fdb0 are related to Matlab code.

lesteve commented:
The decrease of coverage doesn't make any sense given that the diff doesn't add any lines and I didn't touch the tests. I would blame a coveralls glitch.

lesteve commented:
Someone else @rvraghav93 @TomDLT ?

rvraghav93 commented:
I'm not very familiar with that part of the codebase. I think @larsmans may be able to review it better.

But nevertheless do we need a NRT here? Is it possible to conceive one that fails without this fix without using an external dataset?

lesteve commented:
This is a fix coming from upstream so it is debatable whether we should have a non regression test.

I checked that this patch was fixing the segmentation fault in #6687 but I haven't invested the time in putting together a non regression test. I don't really know this code either, so it may take a while ...

rvraghav93 commented:
> This is a fix coming from upstream so it is debatable.

Indeed... I think it's not feasible to include such tests.

lesteve commented:
> Indeed... I think it's not feasible to include such tests.

Then any strong objection on adding your +1 so that this one can be merged?

rvraghav93 commented:
Nothing from my side! This looks good to me. Thanks for the fix @lesteve

lesteve commented:
> Nothing from my side! This looks good to me. Thanks for the fix @lesteve

@rvraghav93 can this be merged then? @ogrisel gave his +1 already.

raghavrv commented:
@agramfort merge?

agramfort commented:
thx @lesteve
